lua/starfall/libs_sh/players.lua

```diff
@@ -514,6 +545,214 @@ if CLIENT then
 		
 		ply:AnimSetGestureWeight(slot, weight)
 	end
+	
+	--- Plays an animation on the player
+	-- @client
+	-- @param sequence Sequence number or string name
+	-- @param progress Optional float (Default 0), the progress of the animation. Ranging from 0-1
+	-- @param rate Optional float (Default 1), the playback rate of the animation
+	-- @param loop Optional bool (Default false), should the animation loop
+	-- @param auto_advance Optional bool (Default true), should the animation handle advancing itself
+	-- @param act Optional number or string name (Default sequence value), the activity the player should use
+	function player_methods:setAnimation(seq, progress, rate, loop, auto_advance, act)
+		local ply = getply(self)
+		checkpermission(instance, ply, "entities.setPlayerRenderProperty")
+		
+		if isstring(seq) then
```

Comment:
Can you start making checktypes a single line for each param? I'll fix the ones from prior prs later as well.